src/main/scala/Core/CSR.scala

```diff
@@ -463,7 +474,7 @@ class CSR(implicit val conf: FlexpretConfiguration) extends Module {
       reg_prv1 := reg_prv
       reg_ie1 := reg_ie
       // privileged mode with interrupts disabled
-      reg_prv := 3.U(2.W)
+      reg_prv := RegInit(VecInit(Seq.fill(conf.threads) { 3.U(2.W) }))
```

Comment:
I am not sure this is what you want? A `RegInit` initializes the register once, not every time `io.exception` is true. 


Also, are all threads promoted to privileged mode when an exception occurs? (Is there a difference between exception and interrupt here?). Doesnt it make more sense to promote only a single thread? The original code writes a singular value into a vector. It should be equivalent of setting only t0 privilege to 3, and all other threads to no privilege

Reply:
When looking more into the FP code. It seems that these are execptions from CPU faults, not from IE, EE or external interrupts. 

Will an exception cause ALL threads to jump to the exception handler? Or only the thread that caused it?

Reply:
I am not quite sure what should be the case here. All I know is that how the code used to be did not build. I just tried to understand what the intention seemed to be and tried to fix the error so it would at least build.

But I changed it away from RegInit at least